### ISSUE TITLE
Strip LogRecord.args Attribute After foreign_pre_chain Processors Run in ProcessorFormatter

### ISSUE THREAD

**Issue by jonathanmeier5** [hynek/structlog#227]
First off, thank you to everyone involved in maintaining this project!

I love the chained approach it provides to building up a log message context. Great work!

I want to start using a structlog-based formatter to output logs as json for several small to medium size code bases I manage.

In particular, I want to be able to write a `processor` that takes the *keys* from the `LogRecord.args` attribute and insert them into my `event_dict`.

Right now, when the `ProcessorFormatter` class receives a log record from `logging` rather than `structlib`, it sets `LogRecord.args = ()` after generating a message [here](https://github.com/hynek/structlog/blob/0e209d95652c0f9b2943bf0334d74f4964fc7f13/src/structlog/stdlib.py#L519).

This means that none of my downstream processors defined in `foreign_pre_chain` can work with the original `LogRecord.args` keys.

Would the maintainers be open to moving the `record.args = ()` setting to *after* the `foreign_pre_chain` processors run? I have a working version locally and can add a test if it's a PR you would accept.

Speaking more broadly, would you be open to me adding some kind of `migration` docs for projects that currently use the stdlib `logging` package but would like to switch to `structlog`?

**Comment by hynek** [comment]
Oof, so it took me a lot of effort to not procrastinate on the answer for a year like I usually do, because all this code we talk about here has been written by other people who are more intimate with stdlib logging. But since it never helped, here we go! ;)

Generally speaking, we can't change the default behavior but maybe it's possible to add an option?

---

> Speaking more broadly, would you be open to me adding some kind of migration docs for projects that currently use the stdlib logging package but would like to switch to structlog?

I'm _always_ open to docs contributions. :) Especially to the murky stdlib parts that I have to read up on myself every time I need to change something.

**Comment by jonathanmeier5** [comment]
Thanks for the quick reply. :) I saw today that you're giving a talk at PyGotham on maintaining open source projects when it's not your job. Seems like you have a lot of experience with this. Looking forward to the talk.

I opened a PR for the `args` issue. I'm not thrilled with the keyword arg name I chose but I couldn't think of anything better.

I'm actually giving a lightning talk tomorrow on moving to structlog. I'll try to consolidate my thoughts and add some migration-oriented docs as a follow on.

Thanks again!
